api/cpp/platform.rs

```diff
@@ -185,6 +185,14 @@ pub unsafe extern "C" fn slint_window_adapter_new(
         set_position,
     });
 
+    if let Some(scale_factor) =
+        option_env!("SLINT_SCALE_FACTOR").and_then(|x| x.parse::<f32>().ok()).filter(|f| *f > 0.)
```

Comment:
Isn't that a little dangerous for regular C++ users? Or is it safe because it'll be overridden anyway later when a windowing system is used?

I guess ideally this would be a cmake cache variable that - if set - maps to an environment variable set for corrosion to pass to cargo. Maybe the cache variable could still be called `SLINT_SCALE_FACTOR` but the environment variable would be `CMAKE_SLINT_SCALE_FACTOR`.

Reply:
If the C++ programmer set a C++ scale factor at runtime, it will be overridden.
But you're right that this probably should be a cmake setting rather than a env variable.